### PR TITLE
Update free-programming-books-subjects.md

### PR DESCRIPTION
**Pull request by PriyanshuSahani**
Added a book pdf on Data structures with C programming language.

## What does this PR do?
Add resource(s) | Remove resource(s) | Add info | Improve repo
Add resource

### Why is this valuable (or not)?
C is widely used for programming close to the hardware so learning DS with C is a great investment

### How do we know it's really free?
No email is required

### For book lists, is it a book? For course lists, is it a course? etc.
Its a book

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [x] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction).
- [x] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

**Review by PriyanshuSahani — COMMENTED** [review]
Added PDF indication on "Data Structures With C"

File: books/free-programming-books-subjects.md [hunk 1]
```diff
@@ -108,6 +108,7 @@ Books that cover a specific programming language can be found in the  [BY PROGRA
 * [Data Structures and Algorithms Tutorial](https://www.tutorialspoint.com/data_structures_algorithms/) - Tutorials Point (HTML, PDF)
 * [Data Structures Succinctly Part 1, Syncfusion](https://www.syncfusion.com/resources/techportal/ebooks/datastructurespart1) (PDF, Kindle) (email address *requested*, not required)
 * [Data Structures Succinctly Part 2, Syncfusion](https://www.syncfusion.com/resources/techportal/ebooks/datastructurespart2) (PDF, Kindle) (email address *requested*, not required)
+* [Data Structures With C](https://www.fresources.tech/dtu/coe/Data%20Structures/file?fileId=https%3A%2F%2Fdata-storage.sgp1.digitaloceanspaces.com%2FDTU%2FDS%2520with%2520C%2520-%2520By%2520Schaum.pdf) - Seymour Lipschutz (PDF)
```
Comment (shrn01) [inline books/free-programming-books-subjects.md:111]: Is this a free book?

Comment (PriyanshuSahani) [reply books/free-programming-books-subjects.md:111]: Yes it is.
The link does not require any email either.

**Comment by eshellman** [comment]
This appears not to be an authorized distribution. Please see docs/CONTRIBUTING.md guidelines.

**Pull request closed without merging**